tools/api-markdown-documenter/src/MarkdownDocumenter.ts

```diff
@@ -0,0 +1,159 @@
+ */
+export function renderDocuments(
+    partialConfig: MarkdownDocumenterConfiguration,
+): MarkdownDocument[] {
+    const config = markdownDocumenterConfigurationWithDefaults(partialConfig);
+    const apiModel = config.apiModel;
+
+    console.log(`Rendering markdown documentation for API Model ${apiModel.displayName}...`);
+
+    const documents: MarkdownDocument[] = [];
+
+    // Always render Model document
+    documents.push(renderModelDocument(apiModel, config));
+
+    const filteredPackages = apiModel.packages.filter(
+        (apiPackage) => !config.packageFilterPolicy(apiPackage),
+    );
+    if (filteredPackages.length !== 0) {
+        // For each package, walk the child graph to find API items which should be rendered to their own document
+        // per provided policy.
+
+        for (const packageItem of filteredPackages) {
+            // Always render documents for packages under the model
+            documents.push(renderPackageDocument(packageItem, config));
+
+            const packageEntryPoints = packageItem.entryPoints;
+            if (packageEntryPoints.length !== 1) {
+                throw new Error(
+                    `Encountered multiple EntryPoint items under package "${packageItem.name}". ` +
+                        "API-Extractor only supports single-entry packages, so this should not be possible.",
+                );
+            }
+
+            const packageEntryPointItem = packageEntryPoints[0];
+
+            const packageDocumentItems = getDocumentItems(packageEntryPointItem, config);
+            for (const apiItem of packageDocumentItems) {
+                documents.push(renderApiItemDocument(apiItem, config));
+            }
+        }
+    }
+
+    console.log("Documents rendered.");
+
+    return documents;
+}
+
+/**
+ * Renders the provided model and its contents, and writes each document to a file on disk.
+ *
+ * @remarks
+ * Which API members get their own documents and which get written to the contents of their parent is
+ * determined by {@link PolicyOptions.documentBoundaries}.
+ *
+ * The file paths under which the files will be saved is determined by the provided output path and the
+ * following configuration properties:
+ *
+ * - {@link PolicyOptions.documentBoundaries}
+ * - {@link PolicyOptions.hierarchyBoundaries}
+ *
+ * @param apiModel - The API model being processed.
+ * This is the output of {@link https://api-extractor.com/ | API-Extractor}.
+ * @param partialConfig - A partial {@link MarkdownDocumenterConfiguration}.
+ * Missing values will be filled in with defaults defined by {@link markdownDocumenterConfigurationWithDefaults}.
+ * @param maybeMarkdownEmitter - The emitter to use for generating Markdown output.
+ * If not provided, a {@link MarkdownEmitter | default implementation} will be used.
+ */
+export async function renderFiles(
+    partialConfig: MarkdownDocumenterConfiguration,
+    outputDirectoryPath: string,
+    maybeMarkdownEmitter?: MarkdownEmitter,
```

Comment:
The only real benefit is signaling to the reader that we don't necessarily have a proper emitter yet. But since we assign it on the 3rd line of the function, it probably isn't particularly valuable. I'll clean this up.